iothub/device/src/Transport/AmqpIoT/AmqpIoTReceivingLink.cs

```diff
@@ -272,67 +272,57 @@ private void OnTwinChangesReceived(AmqpMessage amqpMessage)
             {
                 _receivingAmqpLink.DisposeDelivery(amqpMessage, true, AmqpIoTConstants.AcceptedOutcome);
                 string correlationId = amqpMessage.Properties?.CorrelationId?.ToString();
-                int status = getStatus(amqpMessage);
+                int status = GetStatus(amqpMessage);
 
                 if (status >= 400)
                 {
                     // Handle failures
-                    _onDesiredPropertyReceived.Invoke(null, correlationId, null);
+                    _onTwinMessageReceived.Invoke(null, correlationId, null);
                     string error = null;
-                    using (StreamReader reader = new StreamReader(amqpMessage.BodyStream, System.Text.Encoding.UTF8))
-                    {
-                        error = reader.ReadToEnd();
-                    }
+                    var reader = new StreamReader(amqpMessage.BodyStream, System.Text.Encoding.UTF8);
+                    error = reader.ReadToEnd();
+                    reader.Dispose();
+                    // Retry for Http status code request timeout, Too many requests and server errors
+                    throw new IotHubException(error, status >= 500 || status == 429 || status == 408);
+                }
 
-                    if (status >= 500 || status == 429 || status == 408)
-                    {
-                        throw new IotHubException(error, true);
-                    }
-                    else
-                    {
-                        throw new IotHubException(error, false);
-                    }
+                Twin twin = null;
+                TwinCollection twinProperties = null;
+
+                if (correlationId == null)
+                {
+                    // Here we are getting desired property update notifications and want to handle it first
+                    var reader = new StreamReader(amqpMessage.BodyStream, System.Text.Encoding.UTF8);
```

Comment:
Why not use `using` here instead of the dispose call on 298?

Reply:
On this line I can use "using" and have made changes to address this.

On line 282 however, it shows warning since we throw an exception at line 285 so that still needs to be disposed